packages/server/api/src/app/flows/step-run/step-run-progress.handler.ts

```diff
@@ -0,0 +1,81 @@
+import { 
+    ExecutioOutputFile, 
+    isNil, 
+    ProjectId, 
+    StepOutputStatus, 
+    StepRunResponse,
+    WebsocketClientEvent,
+} from '@activepieces/shared'
+import { FastifyBaseLogger } from 'fastify'
+import { Socket } from 'socket.io'
+import { fileService } from '../../file/file.service'
+import { flowRunService } from '../flow-run/flow-run-service'
+
+function createProgressHandler(context: StepExecutionContext) {
+    return async (progressEvent: FlowRunProgressEvent): Promise<void> => {
+        const { socket, projectId, logger, stepName, requestId } = context
+        
+        if (progressEvent.runId !== context.runId) {
+            return
+        }
+
+        try {
+            const flowRun = await flowRunService(logger).getOneOrThrow({
+                id: progressEvent.runId,
+                projectId,
+            })
+            
+            if (isNil(flowRun.logsFileId)) {
+                return
+            }
+
+            const { data: fileData } = await fileService(logger).getDataOrThrow({
+                fileId: flowRun.logsFileId,
+                projectId: flowRun.projectId,
+            })
+            
+            const executionData: ExecutioOutputFile = JSON.parse(fileData.toString('utf8'))
+            const stepOutput = executionData.executionState.steps[stepName]
```

Comment:
That's wrong, if the trigger is inside the loop, then it would be nested there. I think we have a function that find the step result inside executionDate in the frontend